### PR TITLE
Link to new FSM diagram in developer README

### LINKED ISSUE #2294: Update existing emails and request FSM diagram to match current architecture
### Issue description

We have a domain request FSM diagram already but it's out of date. We should update it with our current FSM logic and email sending logic

### Acceptance criteria

- [ ] domain request state diagram now represents our current system

### Additional context

voted #2 most import thing for us to diagram in eng huddle

### Links to other issues

_No response_

### PR DESCRIPTION
**Pull request by Matt-Spence**
Resolves #2294 

## Changes

- Adds link to the [new Request flow FSM diagram](https://miro.com/app/board/uXjVMuqbLOk=/?moveToWidget=3458764594819017396&cot=14) in developer README

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

**Review by abroddrick — COMMENTED** [review]
Trying to get Rachid and Erin to look over the diagram now. Will Approve tomorrow if they have no comments

**Comment by erinysong** [comment]
Copying over a comment from the Miro here to reduce tabs:
FYR: Application is an outdated term we used to describe domain requests. Replaced references to it since it's a quickfix but just wanted to give a heads up if you see "domain application" anywhere, it most likely needs to be updated to "domain request"

File: docs/developer/README.md [hunk 1]
```diff
@@ -357,4 +357,7 @@ Then, copy the variables under the section labled `s3`.
 1. On the app, navigate to `\admin`.
 2. Under models, click `Waffle flags`.
 3. Click the `disable_email_sending` record. This should exist by default, if not - create one with that name.
-4. (Important) Set the field `everyone` to `Yes`. This field overrides all other settings 
\ No newline at end of file
+4. (Important) Set the field `everyone` to `Yes`. This field overrides all other settings 
+
+## Request Flow FSM Diagram
+There is a diagram detailing the flow of domain requests and resulting domain objects [here](https://miro.com/app/board/uXjVMuqbLOk=/?moveToWidget=3458764594819017396&cot=14)
```
Comment (erinysong) [inline docs/developer/README.md:363]:
```suggestion

The [.gov Domain Request & Domain Status Digram](https://miro.com/app/board/uXjVMuqbLOk=/?moveToWidget=3458764594819017396&cot=14) visualizes the domain request flow and resulting domain objects.

```

Comment (erinysong) [reply docs/developer/README.md:363]: Generally when making hyperlinks, you want your linked text to describe what page you're linking to
([plain text language guide on links](https://www.plainlanguage.gov/guidelines/web/write-effective-links/) if you're interested in further reading)

**Review by erinysong — CHANGES_REQUESTED** [review]
State diagram looks good - left a comment for reference + small content request but can approve once that's done!

File: docs/developer/README.md (hunk 1 above)
Comment (abroddrick) [reply docs/developer/README.md:363]: Good call out Erin!

**Review by erinysong — APPROVED** [review]

**Review by abroddrick — APPROVED** [review]

**Pull request merged**